strawberry/annotation.py

```diff
@@ -228,7 +184,7 @@ def create_union(self, evaled_type) -> StrawberryUnion:
     @classmethod
     def _is_async_type(cls, annotation: type) -> bool:
         origin = getattr(annotation, "__origin__", None)
-        return origin in ASYNC_TYPES
+        return isinstance(origin, type) and issubclass(origin, ASYNC_TYPES)
```

Comment:
how come this was changed?

Reply:
I remember having an issue with this (which this change fixed) while writing this PR, but reverting this change doesn't seem to produce the error anymore.

Checking for `issubclass` have an advantage here that it would work for custom implementations of those types/abcs as well, but knowing now that the change is not actually required, I'm fine in reverting it if you prefer :)

Reply:
let's revert 😊 we can reupdate it later if needed

Reply:
Reverted! :)